From 72903dbb208adf72be1d877816a3d55e37ac4588 Mon Sep 17 00:00:00 2001
From: Ulf Samuelsson <ulf.samuelsson@atmel.com>
Date: Sat, 16 Apr 2011 00:44:05 +0200
Subject: [PATCH 04/24] at91_mci: Do a reset after each request.

My AT91SAM9260 board + Marvell 8686 SDIO WLAN adater are not happy without
applying the MCI reset trick.

Signed-off-by: Ville Syrjala <syrjala@sci.fi>
Signed-off-by: Nicolas Ferre <nicolas.ferre@atmel.com>
Ported to linux-2.6.38
Signed-off-by: Ulf Samuelsson <ulf.samuelsson@atmel.com>
---
 drivers/mmc/host/at91_mci.c |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/mmc/host/at91_mci.c b/drivers/mmc/host/at91_mci.c
index d3e6a96..5955d7c 100644
--- a/drivers/mmc/host/at91_mci.c
+++ b/drivers/mmc/host/at91_mci.c
@@ -624,11 +624,10 @@ static void at91_mci_process_next(struct at91mci_host *host)
 		at91_mci_send_command(host, host->request->stop);
 	} else {
 		del_timer(&host->timer);
-		/* the at91rm9200 mci controller hangs after some transfers,
+		/* the mci controller hangs after some transfers,
 		 * and the workaround is to reset it after each transfer.
 		 */
-		if (cpu_is_at91rm9200())
-			at91_reset_host(host);
+		at91_reset_host(host);
 		mmc_request_done(host->mmc, host->request);
 	}
 }
-- 
1.6.3.3

